### PR TITLE
requirements: correct setup.py install_requires and add tooling to freeze reqs

### PR DESCRIPTION
**Pull request by cjp256**
- Add tools/freeze-requirements.sh script to automatically update
  requirements.txt and requirements-devel.txt.

- Add make target to freeze-requirements.

- Update setup.py install_requires list and extend with [dev] extras.

- Set pinned pip version in environment-setup-local.sh.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,13 +2,17 @@
...
+	tools/freeze-requirements.sh
+
 .PHONY: test-black
 test-black:
 	black --check --diff .
 
 .PHONY: test-codespell
 test-codespell:
-	codespell --quiet-level 4 --ignore-words-list keyserver --skip '*.tar,*.xz,*.zip,*.bz2,*.7z,*.gz,*.deb,*.rpm,*.snap,*.gpg,*.pyc,*.png,*.ico,*.jar,changelog,.git,.hg,.mypy_cache,.tox,.venv,_build,buck-out,__pycache__,build,dist,.vscode,parts,stage,prime,test_appstream.py,./snapcraft.spec,./.direnv'
+	codespell --quiet-level 4 --ignore-words-list crate,keyserver --skip '*.tar,*.xz,*.zip,*.bz2,*.7z,*.gz,*.deb,*.rpm,*.snap,*.gpg,*.pyc,*.png,*.ico,*.jar,changelog,.git,.hg,.mypy_cache,.tox,.venv,_build,buck-out,__pycache__,build,dist,.vscode,parts,stage,prime,test_appstream.py,./snapcraft.spec,./.direnv'
```
Comment (cmatsuoka) [inline Makefile:15]: How interesting, codespell thinks "crate" is misspelled "create". Not very rust-friendly it seems.

**Review by cmatsuoka — COMMENTED** [review]
I see that a frozen list of packages is useful to have a deterministic runtime environment, but should the dev frozen environment be added to vcs?

**Comment by cjp256** [comment]
> I see that a frozen list of packages is useful to have a deterministic runtime environment, but should the dev frozen environment be added to vcs?

It ensures CI doesn't break.  Ideally we wouldn't be pinning any dev dependencies in setup.py, but I did for the moment for the the linters that otherwise break.

**Review by cmatsuoka — APPROVED** [review]
Ideally we should reach a state where we could address warnings as they appear in new versions of the linter tools, but given the size and current state of the code base I agree that it's better to pin versions for now.

**Review by sergiusens — APPROVED** [review]

**Pull request merged**